src/threadlist.cpp

```diff
@@ -644,6 +645,13 @@ ThreadList::waitForAllRestored(Thread *thread)
 
     JTRACE("before DmtcpWorker::postRestart()");
 
+    // A jalloc allocation can result in an mmap call. Although it is rare,
+    // the kernel can place the new region in the free space between the
+    // heap and stack of mtcp_restart i.e., in the restoreBuf region.
+    // It is better to reclaim the restore buffer now before it's too late.
+    // FIXME: can we do any better?
```

Comment:
I like this PR a lot.  I can't imagine a cleaner design.  I suggest removing the FIXME.